samza-core/src/main/java/org/apache/samza/zk/ZkLeaderElector.java

```diff
@@ -114,6 +100,11 @@ public boolean tryBecomeLeader() {
     String predecessor = children.get(index - 1);
     if (!predecessor.equals(currentSubscription)) {
       if (currentSubscription != null) {
+
+        // callback in case if the previous node gets deleted (when previous processor dies)
```

Comment:
That's exactly what I said - I don't know if it is a bug. We just need review this piece to see if can be rewritten without recursive call.